### PR TITLE
dAcTbox_c

### PR DESCRIPTION
**Pull request by robojumper**

### CODE REVIEW

File: include/d/col/cc/d_cc_d.h [hunk 1]
```diff
@@ -83,6 +83,9 @@ class dCcD_Unk : public cCcD_Obj, public cCcD_UnkAttr {
 
     virtual cCcD_ShapeAttr *GetShapeAttr() override;
 
+    // Have to bring these in scope. Consider renaming this
+    // Set function instead or create an inline?
+    using cM3dGUnk::Set;
```
Comment (elijah-thomas774) [inline include/d/col/cc/d_cc_d.h:88]: Using inlines don’t work, cM3dGLin::Set didn’t like this at least. 
Keep with `using`, `.cM3dG::Set`, or we could potentially rename something?

File: include/d/col/c/c_cc_d.h [hunk 2]
```diff
@@ -889,6 +889,9 @@ class cCcD_Obj {
     void OnTgSet() {
         mTg.OnSPrm(1);
     }
+    void OffTgSet() {
+        mTg.OffSPrm(1);
+    }
```
Comment (elijah-thomas774) [inline include/d/col/c/c_cc_d.h:894]: `ClrTgSet` should be the same thing

**Comment by robojumper** [comment]
This is at 92% text, all data matches, 3 functions have very minor issues, `dAcTbox_c::checkForLinkBonk()` is floating point hell

File: include/d/d_cc.h
Comment (elijah-thomas774) [inline include/d/d_cc.h:1]: Sweet, i dont mind this file being named this. Approached this a few times. Only comment is idea of adding a d prefix just to stay consistent with the actors so far

File: include/d/a/obj/d_a_obj_base.h [hunk 3]
```diff
@@ -32,7 +32,11 @@ struct ActorCarryStruct {
...
     /* 0x94 */ f32 field_0x94;
     /* 0x98 */ f32 field_0x98;
-    /* 0x9C */ void *dtor; // ???
+
+    ActorCarryStruct();
+    /* vt 0x9C */
+    virtual ~ActorCarryStruct();
+    // Is this actually part of this struct?
     /* 0xA0 */ u32 field_0xA0;
     /* 0xA4 */ u32 field_0xA4;
```
Comment (elijah-thomas774) [inline include/d/a/obj/d_a_obj_base.h:41]: maybe, I recall needing it at some point. I was carried related (arrow used part of it, loftwing A4 is an actor reference, loftwing used it when carrying player), A0 is a reference to to whatever fn_80358660 constructs

File: src/d/a/obj/d_a_obj_tbox.cpp [hunk 4]
```diff
@@ -0,0 +1,2673 @@
...
+        mDowsingTarget.doUnregister();
+        field_0x11F4 &= ~4;
+    }
+}
+
+extern "C" u16 PARTICLE_RESOURCE_ID_MAPPING_208_;
+
+void dAcTbox_c::fn_8026D950() {
+    fn_800298B0(PARTICLE_RESOURCE_ID_MAPPING_208_, &position, &rotation, nullptr, nullptr, nullptr, nullptr, nullptr);
+}
```
Comment (elijah-thomas774) [inline src/d/a/obj/d_a_obj_tbox.cpp:2377]: some appear particle effect?
image

Comment (elijah-thomas774) [reply src/d/a/obj/d_a_obj_tbox.cpp:2377]: if you are every curious, uploading common.jpc to noclip.website can display a large amount of the particles.

**Review by elijah-thomas774 — APPROVED** [review]

**Pull request merged**